assignments/array-methods.js

```diff
@@ -56,21 +56,40 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
-console.log(fullName);
+    runners.forEach(function(object) {
+        let name=object.first_name+""+object.last_name;
+        fullName.push(name)
+    console.log(fullName);
+})
+
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+const bigLetters = (str) => {
+    return str.toUpperCase();
+}
+allCaps = fullName.map(bigLetters)
 console.log(allCaps); 
 
+
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+const sizeLarge = (object) => {
+    return object.shirt_size === "L"
+}
+largeShirts = runners.filter(sizeLarge)
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+const calc = (acc, donate) =>
+{return acc + donate;}
+runners.forEach(function(object) {
+    ticketPriceTotal.push(object.donation)})
+    ticketPriceTotal = ticketPriceTotal.reduce(calc)
 console.log(ticketPriceTotal);
```

Comment:
Once again, really good job on getting this to work, but you can do it soooooo much faster just doing an anonymous function in the reduce call on the original array.
```suggestion
let ticketPriceTotal = runners.reduce((total, runner) => {
return total + runner.donation
}, 0)
console.log(ticketPriceTotal)
```
See how much shorter this is?